view/view.go

```diff
@@ -0,0 +1,109 @@
+package view
+
+import (
+	"fmt"
+	"html/template"
+	"io/ioutil"
+	"net/url"
+	"path/filepath"
+
+	"go-google-scraper-challenge/helpers/log"
+
+	"github.com/foolin/goview"
+)
+
+const ROOT_VIEW_PATH = "lib/web/views"
+const PARTIAL_PATH = ROOT_VIEW_PATH + "/partials"
+const ICON_PATH = "static/images/icons/"
+
+var viewEngines = map[string]*goview.ViewEngine{}
+
+func SetupView() {
+	defaultEngine := goview.New(getDefaultConfig())
+	viewEngines["default"] = defaultEngine
+	viewEngines["authentication"] = setupNewEngine("authentication")
+
+	goview.Use(defaultEngine)
+}
+
+func SetLayout(layoutName string) {
+	engine := viewEngines[layoutName]
+
+	if engine != nil {
+		engine = setupNewEngine(layoutName)
+	}
+
+	viewEngines[layoutName] = engine
+	goview.Use(engine)
+}
+
+func setupNewEngine(layoutName string) *goview.ViewEngine {
+	newConfig := getDefaultConfig()
+	newConfig.Master = "layouts/" + layoutName
+
+	return goview.New(newConfig)
+}
+
+func getDefaultConfig() goview.Config {
+	return goview.Config{
+		Root:      ROOT_VIEW_PATH,
+		Extension: ".html",
+		Master:    "layouts/default",
+		Partials:  getPartialList(),
+		Funcs: template.FuncMap{
+			"assetsCSS":  assetsCSS,
+			"isActive":   isActive,
+			"renderFile": renderFile,
+			"renderIcon": renderIcon,
+			"urlFor":     urlFor,
```

Comment:
Can we move all of these as helper functions?
It'd be easier to testing 🖖

Reply:
That's a really grate idea, thanks done in [d5592b2](https://github.com/carryall/go-google-scraper-challenge/pull/113/commits/d5592b2d509d486446b8ecdd5c00dca3b989741e) with test added